### PR TITLE
feat(synthetics): update CloudWatch Synthetics NodeJS runtime

### LINKED ISSUE #14153: (synthetics): Add new runtime - syn-nodejs-puppeteer-3.1
AWS CloudWatch Synthetics has released the syn-nodejs-puppeteer-3.1 nodejs runtime.

Adding CDK support for the syn-nodejs-puppeteer-3.1 runtime enables the following features:

https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/CloudWatch_Synthetics_Library_nodejs_puppeteer.html#CloudWatch_Synthetics_runtimeversion-nodejs-puppeteer-3.1

Proposed Solution
Add the synthetics.Runtime.SYNTHETICS_NODEJS_PUPPETEER_3_1 and synthetics.Runtime.SYNTHETICS_NODEJS_PUPPETEER_LATEST runtime to @aws-cdk/aws-synthetics.

- [x]  👋 I may be able to implement this feature request
- [ ] ⚠️ This feature might incur a breaking change

---
This is a 🚀 Feature Request

### PR DESCRIPTION
**Pull request by Khufu-I**
fixes https://github.com/aws/aws-cdk/issues/14153

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Comment by NetaNir** [comment]
Hi @Khufu-I 

Thanks for this PR. Is it really desirable to change the runtime of a deployed `Canaray`, Im not sure I understand the use case. 
Thoughts? I'll tag someone from the synthetics team to get their take on it, too.

**Comment by Khufu-I** [comment]
Thanks @NetaNir for reviewing my CR. The motivation behind automatically updating the runtime of a `Canary` is to keep in sync with Chrome updates. For example AWS consoles only support the latest three versions of Chrome, by using `latestPuppeteer()` the Canary will automatically update to new runtimes on a CDK deployment and verify that there are no breaking changes with new releases of Chrome. This allows website owners to use Synthetics Canary to mimic their customers including the version of Chrome they are using. 

Synthetics Canary `latestPuppeteer` is akin to EC2 machine image [`latestAmazonLinux `](https://github.com/aws/aws-cdk/blob/master/packages/@aws-cdk/aws-ec2/lib/machine-image.ts#L46) which automatically updates on CDK deployment.

**Comment by NetaNir** [comment]
Let's split this PR into two, adding a runtime and the `latest` capability.
The EC2 latest image version are stored in a publicly accessible ssm parameter which gets updated regularly by the EC2 team. Having to maintain "latest" in code will most likely become stale very fast as it relies on a manual update of the code. Without a  mechanism to automatically bump the version of latest we risk misleading customers and adding maintenance overhead.

**Comment by Khufu-I** [comment]
Ok I restricted this change to only `SYNTHETICS_NODEJS_PUPPETEER_3_1` runtime update. I opened another CR #14288 to provide the `latestPuppeteer` helper method.

File: packages/@aws-cdk/aws-synthetics/test/canary.test.ts [hunk 1]
```diff
@@ -44,7 +44,7 @@ test('Canary can have generated name', () => {
       handler: 'index.handler',
       code: synthetics.Code.fromInline('/* Synthetics handler code */'),
     }),
-    runtime: synthetics.Runtime.SYNTHETICS_NODEJS_PUPPETEER_3_0,
+    runtime: synthetics.Runtime.SYNTHETICS_NODEJS_PUPPETEER_3_1,
```
Comment (NetaNir) [inline packages/@aws-cdk/aws-synthetics/test/canary.test.ts:47]: Why was this change if it has no effect on the generated, tested resource?

Comment (Khufu-I) [reply packages/@aws-cdk/aws-synthetics/test/canary.test.ts:47]: I was following the precedence set by the [previous update](https://github.com/aws/aws-cdk/commit/6aac3b6a9bb1586ee16e7a85ca657b544d0f8304#diff-bc3779d8bebfbdd6b6484163cc900f2c772dcd10a6289f059740ef9663fa2b00). Let me know if I should revert this change.

Comment (NetaNir) [reply packages/@aws-cdk/aws-synthetics/test/canary.test.ts:47]: If this value is not tested (not included in the "expect" part), there is no point in changing it.

**Review by NetaNir — APPROVED** [review]

**Pull request merged**